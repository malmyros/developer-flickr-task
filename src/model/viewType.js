module.exports = {
  LARGE: 'LARGE',
  THUMBNAIL: 'THUMBNAIL',
};
