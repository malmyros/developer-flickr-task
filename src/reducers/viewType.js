import { THUMBNAIL } from '../model/viewType';
import { CHANGE_VIEW_TYPE } from '../types';

const viewType = (state = THUMBNAIL, action) => {
  switch (action.type) {
    case CHANGE_VIEW_TYPE:
      return action.viewType;
    default:
      return state;
  }
};

export default viewType;
