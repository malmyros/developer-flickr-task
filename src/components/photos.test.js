import React from 'react';
import { shallow } from 'enzyme';
import Photos from './photos';
import { THUMBNAIL } from '../model/viewType';
import PhotosBuilder from '../builders/photosBuilder';
import PhotoBuilder from '../builders/photoBuilder';

describe('photos', () => {
  it('should render without crashing', () => {
    const photos = new PhotosBuilder.Builder()
      .withPhotos([
        new PhotoBuilder.Builder('1')
          .withFarm(0)
          .withTitle('photo 1')
          .withServer(1)
          .withSecret('secret 1')
          .build(),
        new PhotoBuilder.Builder('2')
          .withFarm(0)
          .withTitle('photo 2')
          .withServer(1)
          .withSecret('secret 2')
          .build(),
      ])
      .build();

    const component = shallow(<Photos viewType={THUMBNAIL} photos={photos.photos}/>);
    expect(component.find('Photo')).toHaveLength(2);
    expect(component).toMatchSnapshot();
  });
});
