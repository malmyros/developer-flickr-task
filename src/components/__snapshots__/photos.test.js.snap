// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`photos should render without crashing 1`] = `
<div
  className="row"
>
  <Photo
    author=""
    description=""
    farm={0}
    id="1"
    key="photo_id_1"
    secret="secret 1"
    server={1}
    title="photo 1"
    viewType="THUMBNAIL"
  />
  <Photo
    author=""
    description=""
    farm={0}
    id="2"
    key="photo_id_2"
    secret="secret 2"
    server={1}
    title="photo 2"
    viewType="THUMBNAIL"
  />
</div>
`;
